sdk/interchain/signature-aggregator.go

```diff
@@ -249,3 +258,305 @@ func (s *SignatureAggregator) Sign(
 		s.quorumPercentage,
 	)
 }
+
+// logWriter is a custom writer that forwards output to a logger
+type logWriter struct {
+	logger logging.Logger
+	level  logging.Level
+}
+
+func (w *logWriter) Write(p []byte) (n int, err error) {
+	// Remove trailing newline if present
+	msg := string(p)
+	if len(msg) > 0 && msg[len(msg)-1] == '\n' {
+		msg = msg[:len(msg)-1]
+	}
+	switch w.level {
+	case logging.Info:
+		w.logger.Info(msg)
+	case logging.Error:
+		w.logger.Error(msg)
+	default:
+		w.logger.Info(msg)
+	}
+	return len(p), nil
+}
+
+// StartSignatureAggregator starts the signature aggregator process.
+// It handles port conflicts and retries if necessary.
+func StartSignatureAggregator(binPath string, configPath string, logger logging.Logger) (int, error) {
+	// Function to check if port is in use
+	isPortInUse := func() bool {
+		conn, err := net.Dial("tcp", "localhost:8080")
+		if err == nil {
+			if err := conn.Close(); err != nil {
+				logger.Warn("Failed to close connection while checking port",
+					zap.Error(err),
+				)
+			}
+			return true
+		}
+		return false
+	}
+
+	// Function to kill existing process
+	killExistingProcess := func() error {
+		// Try pkill first
+		cmd := exec.Command("pkill", "-f", "signature-aggregator")
+		if err := cmd.Run(); err != nil {
+			return fmt.Errorf("failed to kill existing signature-aggregator process: %w", err)
+		}
+		return nil
+	}
+
+	// Try to start the aggregator with retries
+	maxRetries := 3
+	for i := 0; i < maxRetries; i++ {
+		if isPortInUse() {
```

Comment:
i should not base on port, but on a conf file with PID that indicates if the aggregator is executing